ui.frontend/src/view/FormContainer.js

```diff
@@ -39,15 +48,26 @@ export default class FormContainer {
         return id ? this._model.getElement(id) : this._model;
     }
 
+    /**
+     * Returns id of form element present in HTML as parent
+     * @param model
+     * @returns parent id
+     */
+    getParentFormElementId(model) {
+        const parentModel = (model.fieldType && model.repeatable) ? model.parent.parent : model.parent;
+        if (parentModel.fieldType) {
```

Comment:
`if (parentModel.fieldType === 'form')`

Reply:
The intent is to get the element of form and not form.

Reply:
why its in FormContainer? it;'s not making use of form container.
Also, by looking at func impl, it seems it's just getting the id of the parent model of the model passed (which may not be a form).

Reply:
Parent model which is form element that is it has 'field-type' and not a sites container component.

Reply:
IMO. this method should be at field model level. model.parent should return the component automatically with the field type, this logic should not be at view layer.
FormFieldBase.js

```
getParentId() {
       return this.model.parent.id;
}
```

Reply:
this will happen post @rismehta changes

Reply:
will, remove this field-type check

Reply:
removed, since fix is coming from af2-web-runtime